script/check-new-links.sh

```diff
@@ -18,9 +20,11 @@ SKIP_PATTERNS=(
 
 if [ "_${VERBOSE}_" == "__" ]; then VERBOSE=0; fi
 
-for URL in $(git diff develop |
+URLS="$(git diff main |
 		grep '^+' |
-		grep -Eo '(http|https)://[^ )"><]+'); do
+		grep -Eo '(http|https)://[^ )"><]+')"
```

Comment:
> @ixuz [identified](https://github.com/standard-for-public-code/community-implementation-guide-standard/issues/112#issuecomment-2863062050) the for-loop curiously swallowing the error, now it remains to be understood why the checkout doesn't find the `main` branch.

I propose the following adjustment

```suggestion
		grep --extended-regexp --only-matching '(http|https)://[^ )\">\<]+')"
```

With this change, the "check-new-links.sh" script behaves like this:

- When the git diff fails due referencing a non-existant branch, e.g. `"develop"`, then `$ ./script/check-new-links.sh` exits with code 1 (message: `"fatal: ambiguous argument 'develop': unknown revision or path not in the working tree."`)
- Given two valid links in the diff `$ ./script/check-new-links.sh` exits with code 0 (no message output)
- Given two links in the diff, but one is invalid link, then `$ ./script/check-new-links.sh` exits with code 1 (message: `"FAIL: 1 new URLs were not OK"`)
- Given two links in the diff, but both are invalid links, then `$ ./script/check-new-links.sh` exits with code 1 (message: `"FAIL: 2 new URLs were not OK"`)

This should be the correct behavior right? @ericherman